context/authentication.js

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react'
+import * as Google from 'expo-auth-session/providers/google'
+import firebase from 'firebase'
+import * as WebBrowser from 'expo-web-browser'
+
+const firebaseConfig = {
+  apiKey: 'AIzaSyBFiuUulmuQ7Pv1VvxpUQB01AWCEQhIToA',
+  authDomain: 'npm-otp-f6bfc.firebaseapp.com',
+  databaseURL: 'https://npm-otp-f6bfc.firebaseio.com',
+  projectId: 'npm-otp-f6bfc',
+  storageBucket: 'npm-otp-f6bfc.appspot.com',
+  messagingSenderId: '230076165693',
+  appId: '1:230076165693:web:a04f9ad6f64be8ec248454',
+}
+
+if (!firebase.apps.length) {
+  firebase.initializeApp(firebaseConfig)
+}
+
+WebBrowser.maybeCompleteAuthSession()
+
+const AuthenticationContext = React.createContext()
+
+function useFirebaseAuth() {
+  const [user, setUser] = React.useState()
```

Comment:
convention: please prefer the non-qualified names:

```
import React, { useState } from 'react'

// then
const [] = useState()
```

here and everywhere else